src/buoyancy.jl

```diff
@@ -0,0 +1,219 @@
+                                 - b.equation_of_state.β * C.S[i, j, k] )
+
+@inline  thermal_expansion(i, j, k, grid, eos::LinearEquationOfState, C) = eos.α
+@inline haline_contraction(i, j, k, grid, eos::LinearEquationOfState, C) = eos.β
+
+#####
+##### Nonlinear equations of state
+#####
+
+@inline buoyancy_perturbation(i, j, k, grid, b::AbstractBuoyancy{<:AbstractNonlinearEquationOfState}, C) = 
+    - grav(b) * ρ′(i, j, k, grid, b.equation_of_state, C) / b.ρ₀
+
+#####
+##### Roquet et al 2015 idealized nonlinear equations of state
+#####
+
+# Reference: Table 3 in Roquet et el, "Defining a Simplified yet 'Realistic' Equation of State for Seawater", (JPO, 2015)
+roquet_coeffs = Dict(
+                  :linear => (R₀₁₀ = -1.775e-1, R₁₀₀ = 7.718e-1, R₀₂₀ = 0,         R₀₁₁ = 0,          R₂₀₀ = 0,         R₁₀₁ = 0,         R₁₁₀ = 0),
+               :cabbeling => (R₀₁₀ = -0.844e-1, R₁₀₀ = 7.718e-1, R₀₂₀ = -4.561e-3, R₀₁₁ = 0,          R₂₀₀ = 0,         R₁₀₁ = 0,         R₁₁₀ = 0),
+:cabbeling_thermobaricity => (R₀₁₀ = -0.651e-1, R₁₀₀ = 7.718e-1, R₀₂₀ = -5.027e-3, R₀₁₁ = -2.5681e-5, R₂₀₀ = 0,         R₁₀₁ = 0,         R₁₁₀ = 0),
+                :freezing => (R₀₁₀ = -0.491e-1, R₁₀₀ = 7.718e-1, R₀₂₀ = -5.027e-3, R₀₁₁ = -2.5681e-5, R₂₀₀ = 0,         R₁₀₁ = 0,         R₁₁₀ = 0),
+            :second_order => (R₀₁₀ =  0.182e-1, R₁₀₀ = 8.078e-1, R₀₂₀ = -4.937e-3, R₀₁₁ = -2.4677e-5, R₂₀₀ = -1.115e-4, R₁₀₁ = -8.241e-6, R₁₁₀ = -2.446e-3)
+)
+
+"""
+    RoquetIdealizedNonlinearEquationOfState{F, C, T} <: AbstractNonlinearEquationOfState
+
+Parameters associated with the idealized nonlinear equation of state proposed by
+Roquet et al., "Defining a Simplified yet 'Realistic' Equation of State for Seawater", 
+Journal of Physical Oceanography (2015).
+"""
+struct RoquetIdealizedNonlinearEquationOfState{F, C, T} <: AbstractNonlinearEquationOfState
+        ρ₀ :: T
+    coeffs :: C
+end
+
+type_convert_roquet_coeffs(T, coeffs) = NamedTuple{propertynames(coeffs)}(Tuple(T(R) for R in coeffs))
+
+"""
+    RoquetIdealizedNonlinearEquationOfState([T=Float64,] flavor, ρ₀=1025, coeffs=roquet_coeffs[flavor])
+
+Returns parameters for an idealized nonlinear equation of state with reference density 
+`ρ₀`. The `flavor` of the nonlinear equation of state is a symbol corresponding to one of 
+the 5 equations of state in equations (12)--(15) of Roquet et al., "Defining a Simplified 
+yet 'Realistic' Equation of State for Seawater", Journal of Physical Oceanography (2015). 
+The equations of state define the density anomaly `ρ′`, and have the polynomial form
+
+    `ρ′(T, S, D) = Σᵢⱼₐ Rᵢⱼₐ Tⁱ Sʲ Dᵃ`,
+
+where `T` is conservative temperature, `S` is absolute salinity, and `D` is the 
+geopotential depth, currently just `D = -z`. The `Rᵢⱼₐ` are constant coefficients.
+
+Flavors of idealized nonlinear equations of state
+=================================================
+
+    - `:linear`: a linear equation of state, `ρ′ = R₁₀₀ * T + R₀₁₀ * S`
```

Comment:
Should `LinearEquationOfState` just use `RoquetIdealizedNonlinearEquationOfState{:linear}`?

Reply:
I'd rather have users set alpha and beta... ?

Reply:
I think its best to keep these separated, as the Roquet version is a bit more obscure and also has a reference density.